helper/helper/src/main/java/com/linkedin/avroutil1/compatibility/AvroCompatibilityHelper.java

```diff
@@ -814,6 +837,95 @@ private static AvroVersion detectAvroVersion() {
     return AvroVersion.AVRO_1_11;
   }
 
+  private static AvroVersion detectAvroCompilerVersion() {
+    Class<?> specificCompilerClass;
+
+    // There is no avro-compiler in 1.4.X
+    try {
+      specificCompilerClass = Class.forName("org.apache.avro.compiler.specific.SpecificCompiler");
+    } catch (ClassNotFoundException unexpected) {
+      return AvroVersion.AVRO_1_4;
+    }
+
+    // SpecificCompiler.isUnboxedJavaTypeNullable(Schema s) method was added to 1.6.0
+    try {
+      specificCompilerClass.getMethod("isUnboxedJavaTypeNullable", Schema.class);
+    } catch (NoSuchMethodException expected) {
+      return AvroVersion.AVRO_1_5;
+    }
+
+    // In AVRO-1094, a new property was added to velocityEngine in version 1.7.0.
+    // There were no differences in the public interface, so we check for the presence of a property in the private
+    // field, `velocityEngine`.
+    // To further complicate matters, this field is used in 1.7.0 and removed by 1.10.2. So the presence
+    // of the field only indicates that version is one of 1.6, 1.10+.
+    boolean fileResourceLoaderClassExists = checkIfVelocityEngineFileResourceLoaderClassExists(specificCompilerClass);
```

Comment:
this computation should move into the catch block, because in some future >-- 1.11 avro version it might throw, and you only need this value if the version is < 1.8